New OpenSUSE OS: Add OpenSUSE Tumbleweed OS to verify SNP certification.

LakshmiSaiHarika opened the pull request:
- Added OpenSuse Tumbleweed distribution to verify SNP certification process on this Opensuse distribution.
- For the results, please take a look at my forked GH issue for the OpenSUSE Tumbleweed release  [here](https://github.com/LakshmiSaiHarika/snpcert/issues/40)

npmccallum commented:
We need to move to a fixed release. But we can't because Leap 16 isn't available in mkosi yet. See: https://github.com/systemd/mkosi/pull/3994

LakshmiSaiHarika committed:
feat: Add openSUSE Leap 16.0 image configurations

Add host and guest image configurations for openSUSE Leap 16.0 with
required packages including kernel-default, systemd, qemu, and
development tools.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
feat: Add systemd status format configuration module

Add systemd-conf module with combined status format to explicitly add
service name in the service description of journalctl message across all
linux distributions. This was added as a fix on opensuse-leap to get the host and guest service description as per the pattern defined in get_service_description.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
build: Update tools tree and add systemd-conf module

Bump ToolsTreeRelease from 42 to 44 and include the new systemd-conf
module in the common build configuration.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
feat: Add openSUSE Leap support in build modules

Add opensuse-leap to the supported distributions in load-kernel-modules
and network-fix build scripts for proper module configuration and
resolv.conf handling.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
feat: Add openSUSE Leap package mappings for certificate generator

Add opensuse-leap package names for qemu and ovmf in host_os_package.py,
and add zypper package manager support in package_version.sh.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
ci: Update Ubuntu codename from plucky to resolute

Update apt sources to use Ubuntu resolute (26.04) for mkosi installation
in the CI workflow.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
refactor: Add OpenSUSE Leap in the build and release GH workflow

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
fix: Start report-done after sev-certificate-generator.service

Start beacon-report.service after the sev-certificate.txt file is created.
The sev-certificate.txt is generated by the sev-certificate-generator.service.

beacon-report.service raises a `sev-certificate.txt` file not found error
when this service starts after display-guest-logs.service but before
sev-certificate-generator.service.

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika commented:
Please follow the description below for the updated PR changes:
1. Updated the Ubuntu codename from plucky to resolute, since resolute provides mkosi v26 support.
2. Upgraded FedoraToolsTreeRelease from 42 to 44 to ensure compatibility with the latest tool builds using mkosi v26.
3. Added openSUSE Leap 16.0 mkosi configurations for both host and guest images.
4. Configured the systemd status format module to use:
`StatusUnitFormat=combined`
This explicitly includes the service name in journalctl messages across all Linux distributions.
This change was required specifically for openSUSE Leap, where the default format did not match the pattern expected by get_service_description, leading to parsing issues for both host and guest service descriptions.
5. Added openSUSE Leap support to:

    -   kernel module loading scripts
    -   network-fix build scripts
6.  Added package mapping and package manager support for openSUSE Leap:
    -   qemu -> qemu
    -   ovmf → qemu-ovmf-x86_64 
    -    Added zypper package manager support
7. Added openSUSE Leap support in the build and release GitHub workflows.
8. Fixed `report-done.service` ordering issues where beacon-report.service could start before sev-certificate.txt was generated, causing intermittent "file not found" errors.

LakshmiSaiHarika commented on modules/build/common/systemd-conf/mkosi.extra/usr/lib/systemd/system.conf.d/10-status-format.conf:
```diff
@@ -0,0 +1,2 @@
+[Manager]
+StatusUnitFormat=combined
```
This change was required specifically for openSUSE Leap, where the default format did not match the pattern expected by get_service_description (openSUSE didn't include the service name in its description), leading to parsing issues for both host and guest service descriptions.

LakshmiSaiHarika commented on modules/report/host/report-done/mkosi.extra/usr/local/lib/systemd/system/report-done.service:
```diff
@@ -2,8 +2,8 @@
-Requires=display-guest-logs.service sev-certificate-generator.service
-After=display-guest-logs.service sev-certificate-generator.service
+Requires=sev-certificate-generator.service
+After=sev-certificate-generator.service
```
Fixed report-done.service ordering issues where beacon-report.service could start before sev-certificate.txt was generated, causing intermittent "file not found" errors.

<img width="1018" height="767" alt="Image" src="https://github.com/user-attachments/assets/5cd430f1-4e42-418c-9e9a-6613116805c1" />

LakshmiSaiHarika reviewed (commented):
Please refer to my comments for the detailed explanation of the few changes.

LakshmiSaiHarika commented on images/host-opensuse-16.0/mkosi.conf:
```diff
@@ -0,0 +1,43 @@
+        python3-emoji
+        jq
+        avahi
+KernelCommandLine="iommu=nopt"
```
To enable AMD SEV (Secure Encrypted Virtualization) in openSUSE Leap, I had to disable IOMMU passthrough to protect encrypted guest memory integrity as per the OpenSUSE 15 SP6 documentation:
[https://documentation.suse.com/sles/15-SP6/html/SLES-all/cha-vm-security.html](https://documentation.suse.com/sles/15-SP6/html/SLES-all/cha-vm-security.html)

LakshmiSaiHarika reviewed (commented):
Disable IOMMU passthrough on OpenSUSE host in the kernel command line parameter.

LakshmiSaiHarika committed:
refactor: add ovmf file path in OpenSUSE leap

Signed-off-by: Harika Nittala <lnittala@amd.com>

LakshmiSaiHarika committed:
refactor: install ovmf package from Virtualization repository in Opensuse leap

Signed-off-by: Harika Nittala <lnittala@amd.com>
